File: lib_refresh_layout/src/main/java/com/refresh/lib/DaisyFooterView.java
package com.refresh.lib;

import android.animation.ObjectAnimator;
import android.animation.ValueAnimator;
import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.animation.LinearInterpolator;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.github.lib.R;
import com.refresh.lib.contract.PushContract;

/**
 * Description: <DaisyHeaderView><br>
 * Author: gxl<br>
 * Date: 2019/2/25<br>
 * Version: V1.0.0<br>
 * Update: <br>
 */
public class DaisyFooterView extends RelativeLayout implements PushContract {

  private TextView mTxtLoading;
  private ImageView mImgDaisy;
  private ObjectAnimator mRotation;

  public DaisyFooterView(Context context) {
    this(context, null);
  }

  public DaisyFooterView(@NonNull Context context, @Nullable AttributeSet attrs) {
    super(context, attrs);
    initView(context);
  }

  public void initView(Context context) {
    LayoutInflater.from(context).inflate(R.layout.layout_daisy, this);
    mTxtLoading = findViewById(R.id.txt_loading);
    mTxtLoading.setText("上拉加载更多...");
    mImgDaisy = findViewById(R.id.img_daisy);

    mRotation = ObjectAnimator.ofFloat(mImgDaisy, "rotation", 0, 360).setDuration(800);
    mRotation.setRepeatCount(ValueAnimator.INFINITE);
    mRotation.setInterpolator(new LinearInterpolator());

  }
  @Override
  public void onPushEnable(boolean enable) {
    mTxtLoading.setText(enable ? "松开加载" : "上拉加载");;
  }

  @Override
  public void onLoadMore() {
    mTxtLoading.setText("正在加载...");
    mRotation.start();
  }
  public void setLoadMore(boolean b){
    if(b){
      mRotation.start();
    }else{
      mRotation.pause();
    }
  }
}

File: lib_refresh_layout/src/main/java/com/refresh/lib/DaisyFooterView.kt
package com.refresh.lib

import android.animation.ObjectAnimator
import android.animation.ValueAnimator
import android.annotation.SuppressLint
import android.content.Context
import android.os.Build
import android.support.annotation.RequiresApi
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.animation.LinearInterpolator
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import com.github.lib.R

import com.refresh.lib.contract.PushContract

/**
 * Description: <DaisyHeaderView><br>
 * Author: mxdl<br>
 * Date: 2019/2/25<br>
 * Version: V1.0.0<br>
 * Update: <br>
</DaisyHeaderView> */
class DaisyFooterView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) : RelativeLayout(context, attrs), PushContract {

    private lateinit var mTxtLoading: TextView
    private lateinit var mImgDaisy: ImageView
    private lateinit var mRotation: ObjectAnimator

    init {
        initView(context)
    }

    @SuppressLint("ObjectAnimatorBinding")
    fun initView(context: Context) {
        LayoutInflater.from(context).inflate(R.layout.layout_daisy, this)
        mTxtLoading = findViewById(R.id.txt_loading)
        mTxtLoading.text = "上拉加载更多..."
        mImgDaisy = findViewById(R.id.img_daisy)
        mRotation = ObjectAnimator.ofFloat(mImgDaisy, "rotation", 0f, 360f).setDuration(800)
        mRotation.repeatCount = ValueAnimator.INFINITE
        mRotation.interpolator = LinearInterpolator()

    }

    override fun onPushEnable(enable: Boolean) {
        mTxtLoading.text = if (enable) "松开加载" else "上拉加载"
    }

    override fun onLoadMore() {
        mTxtLoading.text = "正在加载..."
        mRotation.start()
    }

    @RequiresApi(Build.VERSION_CODES.KITKAT)
    fun setLoadMore(b: Boolean) {
        if (b) {
            mRotation.start()
        } else {
            mRotation.pause()
        }
    }
}
